ipaplatform/debian/tasks.py

```diff
@@ -27,8 +33,13 @@ def set_nisdomain(nisdomain):
 
     @staticmethod
     def modify_nsswitch_pam_stack(sssd, mkhomedir, statestore):
-        # Debian doesn't use authconfig, this is handled by pam-auth-update
-        return True
+        if mkhomedir:
+            try:
+                ipautil.run(["pam-auth-update", "--package", "--enable", "mkhomedir"])
+            except ipautil.CalledProcessError:
+                return False
+        else:
```

Comment:
I believe you did not mean to have this `else` here since it makes the function return `None` if `mkhomedir` resolves to `True` and the `ipautil.run()` passes.

Reply:
it needs to return True if mkhomedir is not set, but I've added another 'return True' before else: to match your suggestion for restore_pre_ipa_client_configuration above

Reply:
Not sure whether I can't see the change or you did not push it yet, but anyway - just remove the `else:` and the padding in front of `return True` and you'll achieve the same if I am not mistaken.

Reply:
I now pushed what I wrote, missed your latest comment